NetKet/Graph/hypercube.hpp

```diff
@@ -113,6 +127,24 @@ class Hypercube : public AbstractGraph {
     }
   }
 
+  // Edge Colors from users specified map
+  void EdgeColorsFromList(const std::vector<std::vector<int>> &colorlist,
```

Comment:
It looks like `EdgeColorsFromList` and `EdgeColorsFromAdj` are copy-pasted around (or am I missing something?). Wouldn't it be wiser to move the implementation to a common base class?

Reply:
They are in two locations. I considered putting them in a base class, but I didn't want to clutter the AbstractGraph class and abstract class headers all contain only virtual functions. Since the same is done with the distances function, should I just leave these?

Reply:
I think that it is OK to put non-virtual methods in the `Graph` class